docs/details/architecture.md

```diff
@@ -0,0 +1,45 @@
+# Architecture
```

Comment:
This page new but not restructured yet. It is the remnants from the home page and will be expanded with new diagrams.